scripts/o3de/o3de/engine_template.py

```diff
@@ -173,6 +185,18 @@ def _execute_template_json(json_data: dict,
 
         # transform the folder name
         new_dir = _transform(new_dir.as_posix(), replacements, keep_license_text)
+        new_dir = pathlib.Path(new_dir)
+
+        if not keep_restricted_in_instance and 'Platform' in new_dir.parts:
+            try:
+                # the name of the Platform should follow the '/Platform/'
+                pattern = r'/Platform/(?P<Platform>[^/:*?\"<>|\r\n]+/?)'
+                found_platform = re.search(pattern, new_dir.as_posix()).group('Platform')
+                found_platform = found_platform.replace('/', '')
```

Comment:
repeat: This should look over the path parts instead of using regular expressions.
```suggestion
                platform_index = new_dir.parts.index('Platform')
                if platform_index != -1 and platform_index + 1 < len(new_dir.parts):
                    found_platform = new_dir.parts[platform_index + 1]
```